cmd/conduit-pipeline-check/main.go

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	apiv1 "github.com/conduitio/conduit/proto/gen/api/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	exitCodeOk = iota
+	exitCodeErr
+	exitCodeInterrupt
+)
+
+func main() {
+	ctx := cancelOnInterrupt(context.Background())
+
+	exitcode, err := checkPipeline(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	}
+	os.Exit(exitcode)
+}
+
+// checkPipeline returns an exit code depending on the pipeline runtime status.
+// error is returned when an unexpected error is encountered.
+// * missing or invalid flags
+// * gRPC connection failures
+// * operation takes too long to execute
+func checkPipeline(ctx context.Context) (int, error) {
+	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	var (
+		grpcAddress  = flags.String("grpc.address", ":8084", "address of the Conduit server gRPC API")
+		pipelineName = flags.String("pipeline.name", "", "name of the target pipeline")
+		timeout      = flags.Duration("timeout", 10*time.Second, "timeout duration")
+		verbose      = flags.Bool("verbose", false, "return non-zero code on failure instead of logging")
```

Comment:
IIUC `verbose` will make sure a log is printed _in addition_ to the exit code, right? The message seems a bit misleading.

Reply:
ugh yea, will update